modules/consensus/synchronize.go

```diff
@@ -90,8 +90,14 @@ func (cs *ConsensusSet) threadedReceiveBlocks(conn modules.PeerConn) error {
 		}
 
 		// Integrate the blocks into the consensus set.
-		for _, block := range newBlocks {
-			acceptErr := cs.AcceptBlock(block)
+		for i, block := range newBlocks {
+			// Only broadcast the last block.
+			var acceptErr error
+			if !moreAvailable && i == len(newBlocks)-1 {
+				acceptErr = cs.AcceptBlock(block)
+			} else {
+				acceptErr = cs.managedAcceptBlock(block)
+			}
```

Comment:
so... `moreAvailable` depends on an external party. This logic here could actually be abused by an attacker to prevent you from broadcasting a block that you are given, because the attacker could say 'moreAvailable' and then throw an error or something.

Instead, I'm thinking you should do a `defer Broadcast` at the top of the function that will cause you to broadcast the highest block in your consensus set IFF there was at least one successful call to `managedAcceptBlock`.

So maybe something like:

```go
// A comment here explaining what is going on and why
var chainExtended *bool
defer func() {
    if chainExtended {
        recentBlock = someting // there's code in the consensus set to fetch the most recent block. Might need to open a db tx.
        cs.gateway.Broadcast() // would be nice to know that you are using the same broadcast code as AcceptBlock
    }
}()
```

That way you know you are going to broadcast 1 block only, and that you are only going to broadcast if you extend the chain, and you are definitely going to broadcast after extending the chain.